### ISSUE TITLE
Is the independent ide in the plan?

### ISSUE THREAD

**Issue by SherlockGy** [intellij-rust/intellij-rust#4654]

**Comment by Undin** [comment]
At this moment we don't have a plan to create Rust IDE because we can't come up with what additional value it can bring to end users.
Let's imagine potential Rust IDE. It:
* would be typical small IDE (like CLion, PyCharm, etc) with bundled Rust plugin. It wouldn't bring additional quality, extended Rust support and so on. Of course, you wouldn't have to install Rust plugin manually that may be essential for beginners in IntelliJ world.
* must have a debugger. But currently, the native debugger is a paid feature in IntelliJ ecosystem. So this IDE would be also paid
* shouldn't have bundled support of other languages like C++, Python or JavaScript (most popular languages used along with Rust). Of course, in such case, you would be able to install Python and JavaScript plugins but there isn't the corresponding plugin for C/C++. So such IDE loses integration with C++. Otherwise, it would be almost identical with CLion + Rust plugin

So, it would bring almost nothing compared with combination of CLion + Rust plugin (or even would lose something) but require to spend additional resources to support this new product. I think it's more reasonable to use these resource to develop Rust plugin itself

**Comment by cyberhck** [comment]
I agree that's additional resource needed, however in my experience, plugins can't provide nearly as much features as an IDE can.

All the intention items, debugging, deep integration with ecosystem (like goland integrates with goimports, golang-ci, gofmt etc),

I understand and respect the decision, noted :slightly_smiling_face: that was 2019, has anything changed since then?

**Comment by Undin** [comment]
> plugins can't provide nearly as much features as an IDE can
>
> All the intention items, debugging, deep integration with ecosystem (like goland integrates with goimports, golang-ci, gofmt etc),

There is no technical reason why a plugin can't provide the same features as IDE. Moreover, almost all IntelliJ IDEs (including GoLand) consist of some core part and the corresponding language plugin(s) under the hood, and such plugins can be installed in other IDEs (for example, [Go plugin](https://plugins.jetbrains.com/plugin/9568-go) in IntelliJ IDEA Ultimate) and brings the same functionality to that IDE.
All mentioned features just should be implemented and the creation of new IDE won't create these features from anything, unfortunately. BTW Rust plugin has integration with rustfmt as well 😉 

> that was 2019, has anything changed since then?

No, we are still of the same opinion

**Comment by cyberhck** [comment]
while I'm grateful to have this plugin, and yes it has rustfmt, writing rust code after having written C#, golang, TypeScript all in Jetbrains (I love Intellij IDEs), writing Rust now feels like I'm writing code in a measly VSCode. When I type fn it doesn't insert a function like it does in goland, while I know I can do that in IDE settings, I'd have to do this for every single feature that I'd expect from an IDE, I mean your reasoning is solid and it doesn't make sense for jetbrains to invest resources in an IDE.

While I don't know how much effort it takes to get a professional IDE from Jetbrains, but I doubt we'll be seeing what we've seen what Jetbrains can do unfortunately for rust.

Again, Thanks for the quick response, I really appreciate it, I also appreciate the fact that we at least have this plugin, unfortunately that's not the outcome I had hoped :slightly_frowning_face: 

Thanks again.

**Comment by ffleader1** [comment]
> while I'm grateful to have this plugin, and yes it has rustfmt, writing rust code after having written C#, golang, TypeScript all in Jetbrains (I love Intellij IDEs), writing Rust now feels like I'm writing code in a measly VSCode. When I type fn it doesn't insert a function like it does in goland, while I know I can do that in IDE settings, I'd have to do this for every single feature that I'd expect from an IDE, I mean your reasoning is solid and it doesn't make sense for jetbrains to invest resources in an IDE.
> 
> While I don't know how much effort it takes to get a professional IDE from Jetbrains, but I doubt we'll be seeing what we've seen what Jetbrains can do unfortunately for rust.
> 
> Again, Thanks for the quick response, I really appreciate it, I also appreciate the fact that we at least have this plugin, unfortunately that's not the outcome I had hoped 🙁
> 
> Thanks again.

I think the main issue is there is not enough demand. If there is a separated IDE for Rust, I imagine there will be a separated team in charge of managing this IDE, i.e. more resources. Rust, however, is not that popular, or at least not enough, to guarantee such dedication. 

The most relatable example is Go, which has its plugin in 2013 and its own IDE in 2016. Go has a strong demand for it, and go has cemented its place as an awesome languages for building backend and microservices. I use goland on a daily basis. Rust does not have any significant role that you can go: yeah, Rust is "most suitable" for it. Sure, you can argue that Rust is awesome for system programming, but in that area, C/C++ has earned their place for decades. Rust is not much or a frontend language, nor a backend language, nor a game engine language... Rust currently is an alternative for many roles, rather than standing out for association with a particular role. The only notable think I can think of for rust is block chain/coin programing.  

Therefore, the demand for Rust is not high, yet. So, it's just not much of a financial incentive to allocate resource to support a language that does not event know what is it most suitable for, yet. I do love Rust though, so I sure hope the language finds its place.

**Comment by Mersid** [comment]
Well, we got one now! Let's go, RustRover!

**Comment by ffleader1** [comment]
> Well, we got one now! Let's go, RustRover!

Downloading it as we speak

**Comment by Mersid** [comment]
Currently don't see how it's any different than CLion with the plugin, but hopefully as development moves forward we'll get exciting new features that a plugin wouldn't be able to easily implement!
